readthedocs/core/signals.py

```diff
@@ -145,4 +147,14 @@ def add_extra_historical_fields(sender, **kwargs):
         history_instance.extra_history_browser = request.headers.get('User-Agent')
 
 
+@receiver(bind_extra_request_metadata)
+def bind_user_username(request, logger, **kwargs):
+    logger.bind(user_username=getattr(request.user, 'username', ''))
+
+
+@receiver(bind_extra_request_metadata)
+def rewrite_request(request, logger, **kwargs):
+    logger.bind(absolute_url=request.build_absolute_uri())
```

Comment:
These could use docstrings or comments. I assume this is so we can get request data in the response logs?